Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -271,8 +269,8 @@ public struct BSONDocument {
      * Sets a BSON element with the corresponding key
      * if element.value is nil the element is deleted from the BSON
      */
-    internal mutating func set(key: String, to value: BSON?) throws {
-        guard let range = try BSONDocumentIterator.findByteRange(for: key, in: self) else {
+    internal mutating func set(key: String, to value: BSON?) {
```

Comment:
I think we only call this method from the subscript setter, could we make it `private`?